### PR TITLE
Fixes typos in docs

### PR DESCRIPTION
**Pull request by dorzey**

### CODE REVIEW

File: docs/source/index.rst [hunk 1]
```diff
@@ -1,7 +1,7 @@
 📜 The SQL Linter for humans
 ============================
 
-Bored of not having a good SQL linter that works with whichever dialiect
+Bored of not having a good SQL linter that works with whichever dialect
```
Comment (pwildenhain) [inline docs/source/index.rst:4]: Oh boy, right at the top of the front page!

File: docs/source/production.rst [hunk 2]
```diff
@@ -14,7 +14,7 @@ a non-zero code will be returned which can be interrogated to find out
 more.
 
 - At the moment all error states related to linting return *65*.
-- An error as a result of a sqlflSQLFluffuff internal error will return *1*.
+- An error as a result of a SQLFluff internal error will return *1*.
```
Comment (pwildenhain) [inline docs/source/production.rst:17]: Whoops, I bet this made it through in #616 good catch 🧤 ⚾

**Review by pwildenhain — APPROVED** [review]

**Comment by pwildenhain** [comment]
@dorzey #617 should fix these CI failures -- can you update your branch from `master`?

File: docs/source/production.rst (hunk 2 above)
Comment (NiallRees) [reply docs/source/production.rst:17]: This was actually me like an hour ago! Ooops. Missed my double-click!

**Comment by NiallRees** [comment]
I'm going to create an issue to improve the architecture docs. The parser in particular is really quite confusing if you're new to that world, and took me a good while to even partly get my head around, and I imagine quite off-putting to new contributors

**Comment by pwildenhain** [comment]
@NiallRees keep #595 in mind when it comes to updating those docs

**Pull request merged**